openapi-bal-service/src/main/java/io/ballerina/openapi/converter/utils/ConverterCommonUtils.java

```diff
@@ -298,4 +313,75 @@ public static String normalizeTitle(String serviceName) {
+     * @return  boolean output
+     */
+    public static boolean isHttpService(ServiceDeclarationNode serviceNode, SemanticModel semanticModel) {
+        Optional<Symbol> serviceSymbol = semanticModel.symbol(serviceNode);
+        ServiceDeclarationSymbol serviceNodeSymbol = (ServiceDeclarationSymbol) serviceSymbol.get();
+        List<TypeSymbol> listenerTypes = (serviceNodeSymbol).listenerTypes();
+        for (TypeSymbol listenerType : listenerTypes) {
+            if (isHttpListener(listenerType)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static boolean isHttpListener(TypeSymbol listenerType) {
+        if (listenerType.typeKind() == TypeDescKind.UNION) {
+            return ((UnionTypeSymbol) listenerType).memberTypeDescriptors().stream()
+                    .filter(typeDescriptor -> typeDescriptor instanceof TypeReferenceTypeSymbol)
+                    .map(typeReferenceTypeSymbol -> (TypeReferenceTypeSymbol) typeReferenceTypeSymbol)
+                    .anyMatch(typeReferenceTypeSymbol -> isHttpModule(typeReferenceTypeSymbol.getModule().get()));
+        }
+
+        if (listenerType.typeKind() == TypeDescKind.TYPE_REFERENCE) {
+            return isHttpModule(((TypeReferenceTypeSymbol) listenerType).typeDescriptor().getModule().get());
+        }
+        return false;
+    }
+
+    private static boolean isHttpModule(ModuleSymbol moduleSymbol) {
+        if (moduleSymbol.getName().isPresent()) {
+            return HTTP.equals(moduleSymbol.getName().get()) && BALLERINA.equals(moduleSymbol.id().orgName());
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Generate file name with service basePath.
+     */
+    public static String getOpenApiFileName(String servicePath, String serviceName, boolean isJson) {
+        String cleanedServiceName;
+        if (serviceName.isBlank() || serviceName.equals(SLASH) || serviceName.startsWith(SLASH + HYPHEN)) {
+            String[] fileName = serviceName.split(SLASH);
+            // This condition is to handle `service on ep1 {} ` multiple scenarios
+            if (fileName.length > 0 && !serviceName.isBlank()) {
+                cleanedServiceName = FilenameUtils.removeExtension(servicePath) + fileName[1];
+            } else {
+                cleanedServiceName = FilenameUtils.removeExtension(servicePath);
+            }
+        } else if (serviceName.startsWith(HYPHEN)) {
+            // serviceName -> service on ep1 {} has multiple service ex: "-33456"
+            cleanedServiceName = FilenameUtils.removeExtension(servicePath) + serviceName;
+        } else {
+            // Remove starting path separate if exists
+            if (serviceName.startsWith(SLASH)) {
+                serviceName = serviceName.substring(1);
+            }
+            // Replace rest of the path separators with hyphen
+            cleanedServiceName = serviceName.replaceAll(SLASH, "_");
```

Comment:
The comment implies that the path separators should be replaced with hypens, but the code uses underscores. Which one should be the correct behaviour?

Reply:
It should be underscored,  will change the comments :)